tests/components/letpot/test_config_flow.py

```diff
@@ -145,3 +146,128 @@ async def test_flow_duplicate(
+            result["flow_id"],
+            {CONF_PASSWORD: "new-password"},
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "reauth_successful"
+    assert mock_config_entry.data == {
+        CONF_ACCESS_TOKEN: "new_access_token",
+        CONF_ACCESS_TOKEN_EXPIRES: AUTHENTICATION.access_token_expires,
+        CONF_REFRESH_TOKEN: "new_refresh_token",
+        CONF_REFRESH_TOKEN_EXPIRES: AUTHENTICATION.refresh_token_expires,
+        CONF_USER_ID: AUTHENTICATION.user_id,
+        CONF_EMAIL: AUTHENTICATION.email,
+    }
+    assert len(hass.config_entries.async_entries()) == 1
+
+
+@pytest.mark.parametrize(
+    ("exception", "error"),
+    [
+        (LetPotAuthenticationException, "invalid_auth"),
+        (LetPotConnectionException, "cannot_connect"),
+        (Exception, "unknown"),
+    ],
+)
+async def test_reauth_exceptions(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    exception: Exception,
+    error: str,
+) -> None:
+    """Test reauth flow with exception during login and recovery."""
+    mock_config_entry.add_to_hass(hass)
+
+    result = await mock_config_entry.start_reauth_flow(hass)
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "reauth_confirm"
+
+    with patch(
+        "homeassistant.components.letpot.config_flow.LetPotClient.login",
+        side_effect=exception,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_PASSWORD: "new-password"},
+        )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {"base": error}
+
+    # Retry to show recovery.
+    updated_auth = dataclasses.replace(
+        AUTHENTICATION,
+        access_token="new_access_token",
+        refresh_token="new_refresh_token",
+    )
+    with patch(
+        "homeassistant.components.letpot.config_flow.LetPotClient.login",
+        return_value=updated_auth,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_PASSWORD: "new-password"},
+        )
+        await hass.async_block_till_done()
+
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "reauth_successful"
+    assert mock_config_entry.data == {
+        CONF_ACCESS_TOKEN: "new_access_token",
+        CONF_ACCESS_TOKEN_EXPIRES: AUTHENTICATION.access_token_expires,
+        CONF_REFRESH_TOKEN: "new_refresh_token",
+        CONF_REFRESH_TOKEN_EXPIRES: AUTHENTICATION.refresh_token_expires,
+        CONF_USER_ID: AUTHENTICATION.user_id,
+        CONF_EMAIL: AUTHENTICATION.email,
+    }
+    assert len(hass.config_entries.async_entries()) == 1
+
+
+async def test_reauth_different_user_id(
+    hass: HomeAssistant, mock_config_entry: MockConfigEntry
+) -> None:
+    """Test reauth flow with different user ID aborting due to wrong account."""
```

Comment:
While I see how this scenario can be tested, is this possible in the real-world? (Wouldn't only updating the password always return the same account?)

Reply:
See https://github.com/home-assistant/core/pull/135734/files#r1917839184